src/meta/src/hummock/manager/mod.rs

```diff
@@ -878,259 +876,319 @@ impl HummockManager {
+            stats.report_to_metrics(compaction_group_id, self.metrics.as_ref());
+        }
 
-            let mut compact_task_assignment = create_trx_wrapper!(
+        if !trivial_tasks.is_empty() {
+            commit_multi_var!(
+                self.env.meta_store(),
                 self.sql_meta_store(),
-                BTreeMapTransactionWrapper,
-                BTreeMapTransaction::new(&mut compaction.compact_task_assignment,)
-            );
-            compact_task_assignment.insert(
-                compact_task.task_id,
-                CompactTaskAssignment {
-                    compact_task: Some(compact_task.clone()),
-                    context_id: META_NODE_ID, // deprecated
-                },
-            );
+                compaction_statuses,
+                compact_task_assignment,
+                hummock_version_deltas
+            )?;
+            branched_ssts.commit_memory();
 
+            trigger_version_stat(&self.metrics, &current_version);
+            versioning.current_version = current_version;
+            trigger_delta_log_stats(&self.metrics, versioning.hummock_version_deltas.len());
+            self.notify_stats(&versioning.version_stats);
+            self.notify_version_deltas(versioning, last_apply_version_id);
+            self.metrics
+                .compact_task_batch_count
+                .with_label_values(&["batch_trivial_move"])
+                .observe(trivial_tasks.len() as f64);
+            drop(versioning_guard);
+        } else {
             // We are using a single transaction to ensure that each task has progress when it is
             // created.
+            drop(versioning_guard);
             commit_multi_var!(
                 self.env.meta_store(),
                 self.sql_meta_store(),
-                compact_status,
+                compaction_statuses,
                 compact_task_assignment
             )?;
+        }
+        if !pick_tasks.is_empty() {
+            self.metrics
+                .compact_task_batch_count
+                .with_label_values(&["batch_get_compact_task"])
+                .observe(pick_tasks.len() as f64);
+        }
+
+        for compact_task in &mut pick_tasks {
+            let compaction_group_id = compact_task.compaction_group_id;
 
             // Initiate heartbeat for the task to track its progress.
             self.compactor_manager
                 .initiate_task_heartbeat(compact_task.clone());
 
             // this task has been finished.
             compact_task.set_task_status(TaskStatus::Pending);
-
-            trigger_sst_stat(
-                &self.metrics,
-                compaction.compaction_statuses.get(&compaction_group_id),
-                &current_version,
-                compaction_group_id,
-            );
-
-            let compact_task_statistics = statistics_compact_task(&compact_task);
+            let compact_task_statistics = statistics_compact_task(compact_task);
```

Comment:
Prior to this PR, we will call `trigger_sst_stat` here. Is it intentional to remove that?

https://github.com/risingwavelabs/risingwave/blob/e380a5d5651dc6ffa2a7f3f6dce7c29f0f026b32/src/meta/src/hummock/manager/mod.rs#L1126C13-L1126C29